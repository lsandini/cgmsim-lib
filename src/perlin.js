const perlinNoise = require('perlin-noise');
module.exports = function () {
	const time = Date.now();

	const noise = perlinNoise.generatePerlinNoise(288, 1, {
		amplitude: 0.3,
		octaveCount: 3,
		persistence: 0.3,
	});

	const myObject = [];
	let i = 0;
	for (i = 0; i < noise.length; i++) {
		myObject.push({
			noise: noise[i] / 10 - 0.05,
			order: (i),
			time: time + (i) * 1000 * 60 * 5
		});
	}

	return myObject;
};