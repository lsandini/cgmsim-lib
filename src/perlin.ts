import { Perlin, PerlinParams } from "./Types";
import logger from './utils';


const perlinNoise = require('@nickxbs/perlinnoise2');
export default function (params?: PerlinParams): Perlin[] {
	const todayString = new Date().toISOString().substring(0, 10);
	const today = new Date(todayString)
	const defaultParams = {
		amplitude: 0.3,
		octaveCount: 3,
		persistence: 0.3,
		maxAbsValue: 0.05,
		seed: 'cgmsim',
		mode: 'daily'
	}
	const noise = perlinNoise.generatePerlinNoise(288, 1, { ...defaultParams, ...params });

	const maxAbsoluteValue = params?.maxAbsValue ? params.maxAbsValue : defaultParams.maxAbsValue;
	const ratioMaxAbsoluteValue = 0.05 / maxAbsoluteValue;
	const perlinResult: Perlin[] = [];
	for (let i = 0; i < noise.length; i++) {
		perlinResult.push({
			noise: (noise[i] / 10 - 0.05) / ratioMaxAbsoluteValue,
			order: (i),
			time: today.getTime() + (i) * 1000 * 60 * 5
		});
	}
	logger.info('Perlin noise object result: %o', perlinResult);
	return perlinResult;
}