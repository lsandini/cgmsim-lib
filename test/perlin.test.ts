import perlin from "../src/perlin";
describe('test perlin with fake timers', () => {
	beforeAll(() => {
		jest.useFakeTimers('modern');
	});

	beforeEach(()=>{
		jest.setSystemTime(new Date('2001-01-01'));

	})
	afterAll(() => {
		jest.useRealTimers();
	});

	test('run twice in same day should return same array', () => {
		const perl = perlin()
		jest.setSystemTime(new Date('2001-01-01T01:01'));
		const perl2 = perlin()
		expect(perl[0].time).toBe(perl2[0].time)
		expect(perl[0].noise).toBe(perl2[0].noise)
	})
	test('run twice different day should return different array', () => {
		const perl = perlin()
		jest.setSystemTime(new Date('2001-01-02T01:01'));
		const perl2 = perlin()
		expect(perl[0].time).not.toBe(perl2[0].time)
		expect(perl[0].noise).not.toBe(perl2[0].noise)
	})
	
	test('run in same day should return same array', () => {
		const perl = perlin()
		expect(perl.length).toBe(288);
		expect(perl[0]).not.toBe(perl[1])

		expect(perl).toMatchSnapshot()
	})
})

describe('test perlin with real timers', () => {

	test('run twice in same day should return same array', () => {
		const perl = perlin()
		setTimeout(() => {
			const perl2 = perlin()
			expect(perl[0].time).toBe(perl2[0].time)
			expect(perl[0].noise).toBe(perl2[0].noise)
		}, 100);
	})

})