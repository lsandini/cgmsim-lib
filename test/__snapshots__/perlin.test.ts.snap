// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test perlin with fake timers run in same day should return same array 1`] = `
Array [
  Object {
    "noise": -0.0313674934,
    "order": 0,
    "time": 978307200000,
  },
  Object {
    "noise": -0.015428326894964026,
    "order": 1,
    "time": 978307500000,
  },
  Object {
    "noise": 0.0022654622705036015,
    "order": 2,
    "time": 978307800000,
  },
  Object {
    "noise": 0.019117287597122293,
    "order": 3,
    "time": 978308100000,
  },
  Object {
    "noise": 0.032855050499999996,
    "order": 4,
    "time": 978308400000,
  },
  Object {
    "noise": 0.013827385069064752,
    "order": 5,
    "time": 978308700000,
  },
  Object {
    "noise": -0.011061850567625903,
    "order": 6,
    "time": 978309000000,
  },
  Object {
    "noise": -0.016603959280575536,
    "order": 7,
    "time": 978309300000,
  },
  Object {
    "noise": -0.02813357,
    "order": 8,
    "time": 978309600000,
  },
  Object {
    "noise": -0.027781911479136695,
    "order": 9,
    "time": 978309900000,
  },
  Object {
    "noise": -0.028766019568345326,
    "order": 10,
    "time": 978310200000,
  },
  Object {
    "noise": -0.028075262520143886,
    "order": 11,
    "time": 978310500000,
  },
  Object {
    "noise": -0.0313171011,
    "order": 12,
    "time": 978310800000,
  },
  Object {
    "noise": -0.028063327146762594,
    "order": 13,
    "time": 978311100000,
  },
  Object {
    "noise": -0.03334676038633094,
    "order": 14,
    "time": 978311400000,
  },
  Object {
    "noise": -0.02807099992877698,
    "order": 15,
    "time": 978311700000,
  },
  Object {
    "noise": -0.0307845223,
    "order": 16,
    "time": 978312000000,
  },
  Object {
    "noise": -0.013542838658273382,
    "order": 17,
    "time": 978312300000,
  },
  Object {
    "noise": 0.0008499615575539576,
    "order": 18,
    "time": 978312600000,
  },
  Object {
    "noise": 0.014123153099280583,
    "order": 19,
    "time": 978312900000,
  },
  Object {
    "noise": 0.03428883740000002,
    "order": 20,
    "time": 978313200000,
  },
  Object {
    "noise": 0.026769978593525173,
    "order": 21,
    "time": 978313500000,
  },
  Object {
    "noise": 0.024900749374820147,
    "order": 22,
    "time": 978313800000,
  },
  Object {
    "noise": 0.024142833364028768,
    "order": 23,
    "time": 978314100000,
  },
  Object {
    "noise": 0.027080924399999998,
    "order": 24,
    "time": 978314400000,
  },
  Object {
    "noise": 0.011448880563309338,
    "order": 25,
    "time": 978314700000,
  },
  Object {
    "noise": 0.0027610301798561,
    "order": 26,
    "time": 978315000000,
  },
  Object {
    "noise": -0.025555145050359718,
    "order": 27,
    "time": 978315300000,
  },
  Object {
    "noise": -0.049690476500000004,
    "order": 28,
    "time": 978315600000,
  },
  Object {
    "noise": -0.02585286750143885,
    "order": 29,
    "time": 978315900000,
  },
  Object {
    "noise": -0.0132757706647482,
    "order": 30,
    "time": 978316200000,
  },
  Object {
    "noise": 0.015983481415107928,
    "order": 31,
    "time": 978316500000,
  },
  Object {
    "noise": 0.048673526800000005,
    "order": 32,
    "time": 978316800000,
  },
  Object {
    "noise": 0.028853662436690644,
    "order": 33,
    "time": 978317100000,
  },
  Object {
    "noise": 0.008488146979856115,
    "order": 34,
    "time": 978317400000,
  },
  Object {
    "noise": 0.009577198928057556,
    "order": 35,
    "time": 978317700000,
  },
  Object {
    "noise": 0.012567818000000008,
    "order": 36,
    "time": 978318000000,
  },
  Object {
    "noise": 0.01609584147122302,
    "order": 37,
    "time": 978318300000,
  },
  Object {
    "noise": 0.020770331431654673,
    "order": 38,
    "time": 978318600000,
  },
  Object {
    "noise": 0.030637144760431667,
    "order": 39,
    "time": 978318900000,
  },
  Object {
    "noise": 0.04950557300000001,
    "order": 40,
    "time": 978319200000,
  },
  Object {
    "noise": 0.03653936378345325,
    "order": 41,
    "time": 978319500000,
  },
  Object {
    "noise": 0.031182868067625893,
    "order": 42,
    "time": 978319800000,
  },
  Object {
    "noise": 0.03468892920359713,
    "order": 43,
    "time": 978320100000,
  },
  Object {
    "noise": 0.043837882999999994,
    "order": 44,
    "time": 978320400000,
  },
  Object {
    "noise": 0.03857534733597122,
    "order": 45,
    "time": 978320700000,
  },
  Object {
    "noise": 0.03301065364892086,
    "order": 46,
    "time": 978321000000,
  },
  Object {
    "noise": 0.02681377044748201,
    "order": 47,
    "time": 978321300000,
  },
  Object {
    "noise": 0.0240309821,
    "order": 48,
    "time": 978321600000,
  },
  Object {
    "noise": 0.015300650118705031,
    "order": 49,
    "time": 978321900000,
  },
  Object {
    "noise": 0.0045231258661870485,
    "order": 50,
    "time": 978322200000,
  },
  Object {
    "noise": -0.019153768308633093,
    "order": 51,
    "time": 978322500000,
  },
  Object {
    "noise": -0.038569303900000004,
    "order": 52,
    "time": 978322800000,
  },
  Object {
    "noise": -0.011966277439568344,
    "order": 53,
    "time": 978323100000,
  },
  Object {
    "noise": 0.007331684594244603,
    "order": 54,
    "time": 978323400000,
  },
  Object {
    "noise": 0.02476963535971223,
    "order": 55,
    "time": 978323700000,
  },
  Object {
    "noise": 0.04420896399999999,
    "order": 56,
    "time": 978324000000,
  },
  Object {
    "noise": 0.0295831040532374,
    "order": 57,
    "time": 978324300000,
  },
  Object {
    "noise": 0.02252696674820144,
    "order": 58,
    "time": 978324600000,
  },
  Object {
    "noise": 0.014788364143884888,
    "order": 59,
    "time": 978324900000,
  },
  Object {
    "noise": 0.006246910000000001,
    "order": 60,
    "time": 978325200000,
  },
  Object {
    "noise": 0.010441116830935265,
    "order": 61,
    "time": 978325500000,
  },
  Object {
    "noise": 0.021339901646043158,
    "order": 62,
    "time": 978325800000,
  },
  Object {
    "noise": 0.024922301575539582,
    "order": 63,
    "time": 978326100000,
  },
  Object {
    "noise": 0.0376701975,
    "order": 64,
    "time": 978326400000,
  },
  Object {
    "noise": 0.020787349241007197,
    "order": 65,
    "time": 978326700000,
  },
  Object {
    "noise": 0.004509191039568344,
    "order": 66,
    "time": 978327000000,
  },
  Object {
    "noise": -0.017355189863309348,
    "order": 67,
    "time": 978327300000,
  },
  Object {
    "noise": -0.0366762522,
    "order": 68,
    "time": 978327600000,
  },
  Object {
    "noise": -0.024425184046762596,
    "order": 69,
    "time": 978327900000,
  },
  Object {
    "noise": -0.016643586125899286,
    "order": 70,
    "time": 978328200000,
  },
  Object {
    "noise": -0.00869829615971223,
    "order": 71,
    "time": 978328500000,
  },
  Object {
    "noise": -0.005417672000000005,
    "order": 72,
    "time": 978328800000,
  },
  Object {
    "noise": 0.006121437120143888,
    "order": 73,
    "time": 978329100000,
  },
  Object {
    "noise": 0.014270396080575537,
    "order": 74,
    "time": 978329400000,
  },
  Object {
    "noise": 0.023672504905755398,
    "order": 75,
    "time": 978329700000,
  },
  Object {
    "noise": 0.03684512270000001,
    "order": 76,
    "time": 978330000000,
  },
  Object {
    "noise": 0.01151554412949641,
    "order": 77,
    "time": 978330300000,
  },
  Object {
    "noise": -0.010413399061151081,
    "order": 78,
    "time": 978330600000,
  },
  Object {
    "noise": -0.029001231814388493,
    "order": 79,
    "time": 978330900000,
  },
  Object {
    "noise": -0.0470532181,
    "order": 80,
    "time": 978331200000,
  },
  Object {
    "noise": -0.03344253458201439,
    "order": 81,
    "time": 978331500000,
  },
  Object {
    "noise": -0.02233700821870504,
    "order": 82,
    "time": 978331800000,
  },
  Object {
    "noise": -0.0007137247841726563,
    "order": 83,
    "time": 978332100000,
  },
  Object {
    "noise": 0.014732778000000002,
    "order": 84,
    "time": 978332400000,
  },
  Object {
    "noise": 0.014230021535971221,
    "order": 85,
    "time": 978332700000,
  },
  Object {
    "noise": 0.010166942970503606,
    "order": 86,
    "time": 978333000000,
  },
  Object {
    "noise": 0.01399401053956835,
    "order": 87,
    "time": 978333300000,
  },
  Object {
    "noise": 0.022477041200000006,
    "order": 88,
    "time": 978333600000,
  },
  Object {
    "noise": 0.00924866238201439,
    "order": 89,
    "time": 978333900000,
  },
  Object {
    "noise": -0.0039856827949640294,
    "order": 90,
    "time": 978334200000,
  },
  Object {
    "noise": -0.018371724520863303,
    "order": 91,
    "time": 978334500000,
  },
  Object {
    "noise": -0.039254171000000004,
    "order": 92,
    "time": 978334800000,
  },
  Object {
    "noise": -0.022844346787769785,
    "order": 93,
    "time": 978335100000,
  },
  Object {
    "noise": -0.008751799485611514,
    "order": 94,
    "time": 978335400000,
  },
  Object {
    "noise": -0.011248915641726615,
    "order": 95,
    "time": 978335700000,
  },
  Object {
    "noise": -0.018637198500000007,
    "order": 96,
    "time": 978336000000,
  },
  Object {
    "noise": -0.005863556733812947,
    "order": 97,
    "time": 978336300000,
  },
  Object {
    "noise": -0.0012289251438848958,
    "order": 98,
    "time": 978336600000,
  },
  Object {
    "noise": 0.02133957056330936,
    "order": 99,
    "time": 978336900000,
  },
  Object {
    "noise": 0.043735855000000004,
    "order": 100,
    "time": 978337200000,
  },
  Object {
    "noise": 0.032038284460431665,
    "order": 101,
    "time": 978337500000,
  },
  Object {
    "noise": 0.02254480239568346,
    "order": 102,
    "time": 978337800000,
  },
  Object {
    "noise": 0.005559792851079132,
    "order": 103,
    "time": 978338100000,
  },
  Object {
    "noise": -0.005276089999999997,
    "order": 104,
    "time": 978338400000,
  },
  Object {
    "noise": -0.003661796848920859,
    "order": 105,
    "time": 978338700000,
  },
  Object {
    "noise": 0.0009030209496402875,
    "order": 106,
    "time": 978339000000,
  },
  Object {
    "noise": 0.007276128352517973,
    "order": 107,
    "time": 978339300000,
  },
  Object {
    "noise": 0.011539112700000007,
    "order": 108,
    "time": 978339600000,
  },
  Object {
    "noise": 0.010650612379856116,
    "order": 109,
    "time": 978339900000,
  },
  Object {
    "noise": 0.013554570034532379,
    "order": 110,
    "time": 978340200000,
  },
  Object {
    "noise": 0.02440468762014389,
    "order": 111,
    "time": 978340500000,
  },
  Object {
    "noise": 0.03685884569999999,
    "order": 112,
    "time": 978340800000,
  },
  Object {
    "noise": 0.015509785646043162,
    "order": 113,
    "time": 978341100000,
  },
  Object {
    "noise": 0.002634605204316548,
    "order": 114,
    "time": 978341400000,
  },
  Object {
    "noise": -0.016488478758992807,
    "order": 115,
    "time": 978341700000,
  },
  Object {
    "noise": -0.0378086653,
    "order": 116,
    "time": 978342000000,
  },
  Object {
    "noise": -0.03452394806834533,
    "order": 117,
    "time": 978342300000,
  },
  Object {
    "noise": -0.03282469427553957,
    "order": 118,
    "time": 978342600000,
  },
  Object {
    "noise": -0.028069093060431656,
    "order": 119,
    "time": 978342900000,
  },
  Object {
    "noise": -0.030679358200000003,
    "order": 120,
    "time": 978343200000,
  },
  Object {
    "noise": -0.016602333353956833,
    "order": 121,
    "time": 978343500000,
  },
  Object {
    "noise": -0.0005897428741007196,
    "order": 122,
    "time": 978343800000,
  },
  Object {
    "noise": -0.0013884413705035972,
    "order": 123,
    "time": 978344100000,
  },
  Object {
    "noise": 0.005910627000000002,
    "order": 124,
    "time": 978344400000,
  },
  Object {
    "noise": 0.001379425323740996,
    "order": 125,
    "time": 978344700000,
  },
  Object {
    "noise": -0.0014944387064748238,
    "order": 126,
    "time": 978345000000,
  },
  Object {
    "noise": -0.02486928151294964,
    "order": 127,
    "time": 978345300000,
  },
  Object {
    "noise": -0.044407598,
    "order": 128,
    "time": 978345600000,
  },
  Object {
    "noise": -0.02723462437410072,
    "order": 129,
    "time": 978345900000,
  },
  Object {
    "noise": -0.012353604092805753,
    "order": 130,
    "time": 978346200000,
  },
  Object {
    "noise": 0.006842361576978415,
    "order": 131,
    "time": 978346500000,
  },
  Object {
    "noise": 0.03096322400000001,
    "order": 132,
    "time": 978346800000,
  },
  Object {
    "noise": 0.013400180187050373,
    "order": 133,
    "time": 978347100000,
  },
  Object {
    "noise": 0.0017403583597122296,
    "order": 134,
    "time": 978347400000,
  },
  Object {
    "noise": -0.0028307673093525276,
    "order": 135,
    "time": 978347700000,
  },
  Object {
    "noise": -0.015092877799999994,
    "order": 136,
    "time": 978348000000,
  },
  Object {
    "noise": -0.0022932345035971205,
    "order": 137,
    "time": 978348300000,
  },
  Object {
    "noise": 0.006731510064028785,
    "order": 138,
    "time": 978348600000,
  },
  Object {
    "noise": -0.001334800259712221,
    "order": 139,
    "time": 978348900000,
  },
  Object {
    "noise": -0.0051488770999999975,
    "order": 140,
    "time": 978349200000,
  },
  Object {
    "noise": -0.0024295717561151065,
    "order": 141,
    "time": 978349500000,
  },
  Object {
    "noise": 0.005705953316546761,
    "order": 142,
    "time": 978349800000,
  },
  Object {
    "noise": 0.0013694487158273372,
    "order": 143,
    "time": 978350100000,
  },
  Object {
    "noise": -0.006839072499999994,
    "order": 144,
    "time": 978350400000,
  },
  Object {
    "noise": -0.011685687679856116,
    "order": 145,
    "time": 978350700000,
  },
  Object {
    "noise": -0.026513156431654673,
    "order": 146,
    "time": 978351000000,
  },
  Object {
    "noise": -0.022649572110071944,
    "order": 147,
    "time": 978351300000,
  },
  Object {
    "noise": -0.0292993905,
    "order": 148,
    "time": 978351600000,
  },
  Object {
    "noise": -0.01752267361510791,
    "order": 149,
    "time": 978351900000,
  },
  Object {
    "noise": -0.0026021465611510766,
    "order": 150,
    "time": 978352200000,
  },
  Object {
    "noise": -0.006617293780575545,
    "order": 151,
    "time": 978352500000,
  },
  Object {
    "noise": -0.0042184335,
    "order": 152,
    "time": 978352800000,
  },
  Object {
    "noise": -0.016689707499280573,
    "order": 153,
    "time": 978353100000,
  },
  Object {
    "noise": -0.02711520757553957,
    "order": 154,
    "time": 978353400000,
  },
  Object {
    "noise": -0.030878929025179858,
    "order": 155,
    "time": 978353700000,
  },
  Object {
    "noise": -0.043124469000000006,
    "order": 156,
    "time": 978354000000,
  },
  Object {
    "noise": -0.03268850866690648,
    "order": 157,
    "time": 978354300000,
  },
  Object {
    "noise": -0.01888928922230216,
    "order": 158,
    "time": 978354600000,
  },
  Object {
    "noise": -0.024330176665467627,
    "order": 159,
    "time": 978354900000,
  },
  Object {
    "noise": -0.033033118,
    "order": 160,
    "time": 978355200000,
  },
  Object {
    "noise": -0.01760521228776979,
    "order": 161,
    "time": 978355500000,
  },
  Object {
    "noise": -0.004277769017985607,
    "order": 162,
    "time": 978355800000,
  },
  Object {
    "noise": 0.014444465757553954,
    "order": 163,
    "time": 978356100000,
  },
  Object {
    "noise": 0.03314655649999999,
    "order": 164,
    "time": 978356400000,
  },
  Object {
    "noise": 0.017725751590647487,
    "order": 165,
    "time": 978356700000,
  },
  Object {
    "noise": 0.0034574854964028734,
    "order": 166,
    "time": 978357000000,
  },
  Object {
    "noise": -0.01526556679496402,
    "order": 167,
    "time": 978357300000,
  },
  Object {
    "noise": -0.040902602,
    "order": 168,
    "time": 978357600000,
  },
  Object {
    "noise": -0.025880805243884895,
    "order": 169,
    "time": 978357900000,
  },
  Object {
    "noise": -0.016609033684172658,
    "order": 170,
    "time": 978358200000,
  },
  Object {
    "noise": 0.002815875581294966,
    "order": 171,
    "time": 978358500000,
  },
  Object {
    "noise": 0.0197947329,
    "order": 172,
    "time": 978358800000,
  },
  Object {
    "noise": 0.011272021232374113,
    "order": 173,
    "time": 978359100000,
  },
  Object {
    "noise": 0.006566952000000001,
    "order": 174,
    "time": 978359400000,
  },
  Object {
    "noise": -0.00035176115827337956,
    "order": 175,
    "time": 978359700000,
  },
  Object {
    "noise": -0.006583850700000005,
    "order": 176,
    "time": 978360000000,
  },
  Object {
    "noise": -0.002466215169064749,
    "order": 177,
    "time": 978360300000,
  },
  Object {
    "noise": -0.0022691040517985595,
    "order": 178,
    "time": 978360600000,
  },
  Object {
    "noise": 0.008683914636690646,
    "order": 179,
    "time": 978360900000,
  },
  Object {
    "noise": 0.013760957699999993,
    "order": 180,
    "time": 978361200000,
  },
  Object {
    "noise": 0.0038086722302158316,
    "order": 181,
    "time": 978361500000,
  },
  Object {
    "noise": 0.002836301162589934,
    "order": 182,
    "time": 978361800000,
  },
  Object {
    "noise": -0.010330841380575544,
    "order": 183,
    "time": 978362100000,
  },
  Object {
    "noise": -0.025521900000000004,
    "order": 184,
    "time": 978362400000,
  },
  Object {
    "noise": -0.010029600690647483,
    "order": 185,
    "time": 978362700000,
  },
  Object {
    "noise": 0.008404986330215823,
    "order": 186,
    "time": 978363000000,
  },
  Object {
    "noise": 0.01037303649280575,
    "order": 187,
    "time": 978363300000,
  },
  Object {
    "noise": 0.016194698399999985,
    "order": 188,
    "time": 978363600000,
  },
  Object {
    "noise": 0.006267908884892079,
    "order": 189,
    "time": 978363900000,
  },
  Object {
    "noise": 0.003586992058992808,
    "order": 190,
    "time": 978364200000,
  },
  Object {
    "noise": -0.009780377540287773,
    "order": 191,
    "time": 978364500000,
  },
  Object {
    "noise": -0.024188137000000002,
    "order": 192,
    "time": 978364800000,
  },
  Object {
    "noise": -0.012257570147482022,
    "order": 193,
    "time": 978365100000,
  },
  Object {
    "noise": -0.007459728392086334,
    "order": 194,
    "time": 978365400000,
  },
  Object {
    "noise": -0.0038625105374100674,
    "order": 195,
    "time": 978365700000,
  },
  Object {
    "noise": 0.0027299520000000008,
    "order": 196,
    "time": 978366000000,
  },
  Object {
    "noise": 0.006322944420143882,
    "order": 197,
    "time": 978366300000,
  },
  Object {
    "noise": 0.01042917478992806,
    "order": 198,
    "time": 978366600000,
  },
  Object {
    "noise": 0.011737478346043169,
    "order": 199,
    "time": 978366900000,
  },
  Object {
    "noise": 0.010041995300000002,
    "order": 200,
    "time": 978367200000,
  },
  Object {
    "noise": 0.004001852837410079,
    "order": 201,
    "time": 978367500000,
  },
  Object {
    "noise": 0.003187558917985618,
    "order": 202,
    "time": 978367800000,
  },
  Object {
    "noise": 0.0037694476611510805,
    "order": 203,
    "time": 978368100000,
  },
  Object {
    "noise": 0.004187884399999994,
    "order": 204,
    "time": 978368400000,
  },
  Object {
    "noise": -0.005069094320143887,
    "order": 205,
    "time": 978368700000,
  },
  Object {
    "noise": -0.009817010532374097,
    "order": 206,
    "time": 978369000000,
  },
  Object {
    "noise": -0.021851691199280578,
    "order": 207,
    "time": 978369300000,
  },
  Object {
    "noise": -0.0363692815,
    "order": 208,
    "time": 978369600000,
  },
  Object {
    "noise": -0.023750474326618705,
    "order": 209,
    "time": 978369900000,
  },
  Object {
    "noise": -0.016185671140287773,
    "order": 210,
    "time": 978370200000,
  },
  Object {
    "noise": -0.0030160279733812967,
    "order": 211,
    "time": 978370500000,
  },
  Object {
    "noise": 0.012039719999999997,
    "order": 212,
    "time": 978370800000,
  },
  Object {
    "noise": 0.004155023643884891,
    "order": 213,
    "time": 978371100000,
  },
  Object {
    "noise": 0.0010617561194244604,
    "order": 214,
    "time": 978371400000,
  },
  Object {
    "noise": -0.005121839946043177,
    "order": 215,
    "time": 978371700000,
  },
  Object {
    "noise": -0.011286791999999997,
    "order": 216,
    "time": 978372000000,
  },
  Object {
    "noise": 0.004816651639568348,
    "order": 217,
    "time": 978372300000,
  },
  Object {
    "noise": 0.019479166945323756,
    "order": 218,
    "time": 978372600000,
  },
  Object {
    "noise": 0.025661644035971223,
    "order": 219,
    "time": 978372900000,
  },
  Object {
    "noise": 0.04170301700000001,
    "order": 220,
    "time": 978373200000,
  },
  Object {
    "noise": 0.018985533305755392,
    "order": 221,
    "time": 978373500000,
  },
  Object {
    "noise": 0.0024369431928057475,
    "order": 222,
    "time": 978373800000,
  },
  Object {
    "noise": -0.0004319513812949677,
    "order": 223,
    "time": 978374100000,
  },
  Object {
    "noise": -0.0007147569999999964,
    "order": 224,
    "time": 978374400000,
  },
  Object {
    "noise": 0.005399047133093526,
    "order": 225,
    "time": 978374700000,
  },
  Object {
    "noise": 0.01909527778992806,
    "order": 226,
    "time": 978375000000,
  },
  Object {
    "noise": 0.022986606964028794,
    "order": 227,
    "time": 978375300000,
  },
  Object {
    "noise": 0.025301705100000002,
    "order": 228,
    "time": 978375600000,
  },
  Object {
    "noise": 0.009275299633093523,
    "order": 229,
    "time": 978375900000,
  },
  Object {
    "noise": -0.00789207396402878,
    "order": 230,
    "time": 978376200000,
  },
  Object {
    "noise": -0.013860356974100721,
    "order": 231,
    "time": 978376500000,
  },
  Object {
    "noise": -0.017096726800000003,
    "order": 232,
    "time": 978376800000,
  },
  Object {
    "noise": -0.020695544940287768,
    "order": 233,
    "time": 978377100000,
  },
  Object {
    "noise": -0.020067217511510797,
    "order": 234,
    "time": 978377400000,
  },
  Object {
    "noise": -0.03413854480431655,
    "order": 235,
    "time": 978377700000,
  },
  Object {
    "noise": -0.048702973000000004,
    "order": 236,
    "time": 978378000000,
  },
  Object {
    "noise": -0.031860321837410076,
    "order": 237,
    "time": 978378300000,
  },
  Object {
    "noise": -0.012848848201438848,
    "order": 238,
    "time": 978378600000,
  },
  Object {
    "noise": -0.015152726489208637,
    "order": 239,
    "time": 978378900000,
  },
  Object {
    "noise": -0.016197451800000005,
    "order": 240,
    "time": 978379200000,
  },
  Object {
    "noise": -0.018369815130935246,
    "order": 241,
    "time": 978379500000,
  },
  Object {
    "noise": -0.023934728474820147,
    "order": 242,
    "time": 978379800000,
  },
  Object {
    "noise": -0.02871613584172662,
    "order": 243,
    "time": 978380100000,
  },
  Object {
    "noise": -0.040072462,
    "order": 244,
    "time": 978380400000,
  },
  Object {
    "noise": -0.030222003069784172,
    "order": 245,
    "time": 978380700000,
  },
  Object {
    "noise": -0.01878300882589928,
    "order": 246,
    "time": 978381000000,
  },
  Object {
    "noise": -0.012339976928057557,
    "order": 247,
    "time": 978381300000,
  },
  Object {
    "noise": -0.004584958900000005,
    "order": 248,
    "time": 978381600000,
  },
  Object {
    "noise": -0.004043609845323752,
    "order": 249,
    "time": 978381900000,
  },
  Object {
    "noise": 0.003399656910791367,
    "order": 250,
    "time": 978382200000,
  },
  Object {
    "noise": 0.0013491624748201425,
    "order": 251,
    "time": 978382500000,
  },
  Object {
    "noise": 0.004262109500000007,
    "order": 252,
    "time": 978382800000,
  },
  Object {
    "noise": 0.003329136075539571,
    "order": 253,
    "time": 978383100000,
  },
  Object {
    "noise": 0.00970329638489209,
    "order": 254,
    "time": 978383400000,
  },
  Object {
    "noise": 0.005068313012949639,
    "order": 255,
    "time": 978383700000,
  },
  Object {
    "noise": -0.0017846922000000001,
    "order": 256,
    "time": 978384000000,
  },
  Object {
    "noise": 0.005608289781294962,
    "order": 257,
    "time": 978384300000,
  },
  Object {
    "noise": 0.008188846189928066,
    "order": 258,
    "time": 978384600000,
  },
  Object {
    "noise": 0.006246983583453242,
    "order": 259,
    "time": 978384900000,
  },
  Object {
    "noise": 0.010929275000000002,
    "order": 260,
    "time": 978385200000,
  },
  Object {
    "noise": 0.0002700079712230352,
    "order": 261,
    "time": 978385500000,
  },
  Object {
    "noise": -0.01665276811079136,
    "order": 262,
    "time": 978385800000,
  },
  Object {
    "noise": -0.020776594537410077,
    "order": 263,
    "time": 978386100000,
  },
  Object {
    "noise": -0.035277035000000005,
    "order": 264,
    "time": 978386400000,
  },
  Object {
    "noise": -0.03220934538705036,
    "order": 265,
    "time": 978386700000,
  },
  Object {
    "noise": -0.02974217348992806,
    "order": 266,
    "time": 978387000000,
  },
  Object {
    "noise": -0.030348589971942445,
    "order": 267,
    "time": 978387300000,
  },
  Object {
    "noise": -0.0382661081,
    "order": 268,
    "time": 978387600000,
  },
  Object {
    "noise": -0.028816766724460433,
    "order": 269,
    "time": 978387900000,
  },
  Object {
    "noise": -0.019312417886330938,
    "order": 270,
    "time": 978388200000,
  },
  Object {
    "noise": -0.012690198890647486,
    "order": 271,
    "time": 978388500000,
  },
  Object {
    "noise": -0.002933400100000004,
    "order": 272,
    "time": 978388800000,
  },
  Object {
    "noise": 0.007041713661870515,
    "order": 273,
    "time": 978389100000,
  },
  Object {
    "noise": 0.016994007632374103,
    "order": 274,
    "time": 978389400000,
  },
  Object {
    "noise": 0.025622595553956823,
    "order": 275,
    "time": 978389700000,
  },
  Object {
    "noise": 0.043940744200000006,
    "order": 276,
    "time": 978390000000,
  },
  Object {
    "noise": 0.04037854449640288,
    "order": 277,
    "time": 978390300000,
  },
  Object {
    "noise": 0.04446914797553955,
    "order": 278,
    "time": 978390600000,
  },
  Object {
    "noise": 0.04134222611079137,
    "order": 279,
    "time": 978390900000,
  },
  Object {
    "noise": 0.04356756260000001,
    "order": 280,
    "time": 978391200000,
  },
  Object {
    "noise": 0.02331450196762591,
    "order": 281,
    "time": 978391500000,
  },
  Object {
    "noise": 0.0090990354705036,
    "order": 282,
    "time": 978391800000,
  },
  Object {
    "noise": 0.0069295273093525345,
    "order": 283,
    "time": 978392100000,
  },
  Object {
    "noise": 0.00016895000000000104,
    "order": 284,
    "time": 978392400000,
  },
  Object {
    "noise": -0.006841352329496404,
    "order": 285,
    "time": 978392700000,
  },
  Object {
    "noise": -0.017636679818705037,
    "order": 286,
    "time": 978393000000,
  },
  Object {
    "noise": -0.021273396274820142,
    "order": 287,
    "time": 978393300000,
  },
]
`;
